activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -453,6 +453,9 @@ def checkout_and_verify(c)
           c.verify!
         end
         c
+      rescue
+        disconnect!
```

Comment:
It's certainly true that some (well, most) causes of exceptions here will mean all the connections are bad, but I don't see any reason for us to codify that as being strictly true.

For example, individual connections can go bad with some proxy setups.

Mostly, I just don't see what we gain by doing it this way.